lib/octokit/default.rb

```diff
@@ -7,11 +7,6 @@
 require 'octokit/warnable'
 
 if Gem::Version.new(Faraday::VERSION) >= Gem::Version.new('2.0')
-  begin
-    require 'faraday/retry'
-  rescue LoadError
-    Octokit::Warnable.octokit_warn 'To use retry middleware with Faraday v2.0+, install `faraday-retry` gem'
```

Comment:
This warning is the only thing that would even let you know that this gem does retries. There is nothing in the README about this. This would be better as an explicit option to enable retries.

Reply:
So do you suggest disabling retries by default?

Reply:
That would be the best way to remove the warning. The retry functionality needs to be better documented.

Reply:
Added commit 3395ed41c0ee7dbc39856c96599acee181638b24 to remove the retry middleware from the default middleware stack and commit c892eb3ad3e5e32e79c537f94df102947eaa92a2 to provide the retry middleware to insert it into the middleware stack explicitly.